SC45: Wildcard Domain Validation (#269)
cabforum/servercert: docs/BR.md

CAs SHALL NOT perform validations using this method after June 3, 2020. CAs MAY continue to re-use information and validations for domains validated under this method per the applicable certificate data reuse periods.

> **Note**:

Reviewer: Do we need to update this method at all? SC42 passed, so it will not be possible to reuse any method 6 validations starting October 1st, which is before the effective date of this ballot (SC45).

Author: Technically SC42 hasn't passed the IP review period.

It's a little tricky to do the "two versions", but my approach with other ballots has been to "include the language as if it fails" as one commit, and then put another comment "and here's the language if it succeeds", and use two different commit ranges for the ballot.

So yeah, if SC42 succeeds, I think the updated version (and I need to double check the math, but I'm sure you're right) would simply be to "must not be used" this portion because of the SC42 interaction (i.e. we could have done it in SC42 itself, but doing it now will at least ensure there's no ambiguity about wildcards from folks who don't implement/notice SC42)

  * For certificates issued prior to 2021-12-01, the CA MAY also issue Certificates for other FQDNs that end with all the labels of the validated FQDN. This method is suitable for validating Wildcard Domain Names.

>   * For certificates issued on or after 2021-12-01, the CA MUST NOT issue Certificates for other FQDNs that end with all the labels of the validated FQDN unless the CA performs a separate validation for that FQDN using an authorized method. This method is NOT suitable for validating Wildcard Domain Names.

Reviewer 2: "that FQDN" seems to read as though it refers to "the validated FQDN", which doesn't make sense.

Suggestion: s/a separate validation for that FQDN using an authorized method/separate validations for each of those other FQDNs using authorized methods/

Author: So, this is for better or worse the existing language (3.2.2.4.8, 3.2.2.4.20), so I think it might be better to keep symmetry/consistency here, and perhaps clarify this language as follow-up. Do you want to file an issue for that so we can sweep it up in a spring cleanup?

Reviewer 2: Makes sense.  I've filed Issue #272 for this.

1. The CA MUST provide a Random Value unique to the certificate request.
2. The Random Value MUST remain valid for use in a confirming response for no more than 30 days from its creation. The CPS MAY specify a shorter validity period for Random Values, in which case the CA MUST follow its CPS.

**Note**:

>   * For certificates issued prior to 2021-12-01, the CA MAY also issue Certificates for other FQDNs that end with all the labels of the validated FQDN. This method is suitable for validating Wildcard Domain Names.

Reviewer: ```suggestion
  * For Certificates issued prior to 2021-12-01, the CA MAY also issue Certificates for other FQDNs that end with all the labels of the validated FQDN. This method is suitable for validating Wildcard Domain Names.
```

2. The CA MUST verify the Applicant’s control over the .onion Domain Name using at least one of the methods listed below:

   a. The CA MAY verify the Applicant's control over the .onion service by using one of the following methods from [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control):

      i. [Section 3.2.2.4.6 - Agreed-Upon Change to Website](#32246-agreed-upon-change-to-website):

         The CA SHALL only use this method when reusing a previously completed domain validation permitted by that method and [Section 4.2.1](#421-performing-identification-and-authentication-functions). The CA MUST NOT perform new validations using this method.

      ii. [Section 3.2.2.4.18 - Agreed-Upon Change to Website v2](#322418-agreed-upon-change-to-website-v2):

>           The CA SHALL only use this method to validate Wildcard Certificates or as an Authorization Domain Name as specified by that section, except as follows:

Author: Sorry, this was language I'm meaning to remove, but haven't had a chance to pick this up yet :(

>           When the CA performs the validation with the FQDN containing exactly two labels, the hidden service name and "onion", the CA MAY use this validation to validate a Wildcard Domain Name for this FQDN and MAY use this validation to issue Certificates for other FQDNs that end with all the labels, without regard to the prohibition of such certificates in [Section 3.2.2.4.18](#322418-agreed-upon-change-to-website-v2).

Author: So this is a substantial change in the language (although, as mentioned above, it's being removed)

The issue here is that your proposed language changes to allow `foo.bar.onion` to authenticate `*.foo.bar.onion`, and that's explicitly not something that is intended nor good, as covered on the list, the call, and the PR discussion.

But, like I said, this is being removed, so that wildcard authorization/subdomain authorization is only done with an equivalent "proof of control over entire domain namespace" (i.e. the onion hidden service key)

      iii. [Section 3.2.2.4.19 - Agreed-Upon Change to Website - ACME](#322419-agreed-upon-change-to-website-acme):

>            The CA SHALL only use this method to validate Wildcard Certificates or as an Authorization Domain Name as specified by that section, except as follows:

Reviewer: ```suggestion
          Validations for .onion Domain Names SHALL be performed according to [Section 3.2.2.4.19](#322419-agreed-upon-change-to-website-acme). If the method prohibits its use for the validation of Wildcard Domain Names for the .onion Domain Name or other FQDNs that end with the .onion Domain Name and the .onion Domain Name consists of exactly two labels (the hidden service name and "onion"), then the following exception applies:
```

>            When the CA performs the validation with the FQDN containing exactly two labels, the hidden service name and "onion", the CA MAY use this validation to validate a Wildcard Domain Name for this FQDN and MAY use this validation to issue Certificates for other FQDNs that end with all the labels, without regard to the prohibition of such certificates in [Section 3.2.2.4.19](#322419-agreed-upon-change-to-website-acme).

Reviewer: ```suggestion
          The CA MAY use validations performed using this method to validate a Wildcard Domain Name for the validated .onion Domain Name and MAY use validations performed using this method to validate other FQDNs that end with all labels of the validated .onion Domain Name.
```
